contracts/axelar-gateway/src/auth.rs

```diff
@@ -72,6 +73,10 @@ pub fn validate_proof(
         ContractError::InvalidSignatures
     );
 
+    env.storage()
```

Comment:
not very readable to do this within an auth helper. Let's do it in `approve_messages`. That's the entrypoint that gets called frequently.